.github/workflows/e2e-tests-reusable-workflow.yaml

```diff
@@ -0,0 +1,92 @@
+name: e2e test reusable workflow
+
+on:
+  workflow_call:
+    inputs:
+      plugin-test:
+        required: true
+        type: boolean
+      dir-to-test:
+        required: true
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
+          with:
+            submodules: recursive
+
+        - name: Set up Go
+          uses: actions/setup-go@v3
+          with:
+            go-version: v1.22
+
+        - name: Set up gotestfmt
+          uses: gotesttools/gotestfmt-action@v2
+          with:
+            token: ${{ secrets.GITHUB_TOKEN }}
+
+        - name: Setup and start KinD cluster
+          uses: ./.github/workflows/actions/kind
+
+        - name: Build CLI and Add to PATH
+          if: inputs.plugin-test
+          run: |
+            cd ./kubectl-plugin
+
+            go mod download
+            go build -o kubectl-ray -a ./cmd/kubectl-ray.go
+
+            cp ./kubectl-ray /usr/local/bin
+
+        - name: Deploy Kuberay operator
+          id: deploy
+          run: |
+            echo Deploying Kuberay operator
+            cd ray-operator
+
+            IMG="${REGISTRY_ADDRESS}"/kuberay
+            make docker-build -e IMG="${IMG}" -e ENGINE=podman
+            make docker-push -e IMG="${IMG}" -e ENGINE=podman
+
+            make deploy -e IMG="${IMG}"
+            kubectl wait --timeout=90s --for=condition=Available=true deployment -n ray-system kuberay-operator
+
+        - name: Deploy Kuberay Cluster
+          if: inputs.plugin-test
+          run: |
+            echo Deploying Kuberay cluster
+
+            kubectl apply -f ./ray-operator/config/samples/ray-cluster.sample.yaml
+            kubectl wait --timeout=300s --for=condition=ready  pod -l ray.io/cluster=raycluster-sample
+
+        - name: Run e2e tests
+          run: |
+            export KUBERAY_TEST_TIMEOUT_SHORT=1m
+            export KUBERAY_TEST_TIMEOUT_MEDIUM=5m
+            export KUBERAY_TEST_TIMEOUT_LONG=10m
+
+            export KUBERAY_TEST_OUTPUT_DIR=${{ env.TEMP_DIR }}
+            echo "KUBERAY_TEST_OUTPUT_DIR=${KUBERAY_TEST_OUTPUT_DIR}" >> $GITHUB_ENV
+
+            set -euo pipefail
+            cd ${{ inputs.dir-to-test }}
+            go test -timeout 30m -v ./test/e2e -json 2>&1 | tee ${KUBERAY_TEST_OUTPUT_DIR}/gotest.log | gotestfmt
+
+        - name: Print KubeRay operator logs
+          if: (!inputs.plugin-test)  && always() && steps.deploy.outcome == 'success'
```

Comment:
So my reasoning is that since the the kubectl plugin test isn't testing or using kuberay operator, there isn't a reason to print out the operator logs. Willing to discuss more if you disagree.

Reply:
Understood. It's okay not to print the log. BTW I just found that there are two spaces after `(!inputs.plugin-test)`. Could you replace it with single space?

Reply:
Done! Thanks!